### PR TITLE
When encountering damaged data, continue to compress other data.

### PR DESCRIPTION
**Pull request by mikechengwei**
Optimize: #4046 

## Changes

When encountering damaged data, continue to compress other data.

### CODE REVIEW

File: pkg/compact/compact.go [hunk 1]
```diff
@@ -728,7 +728,8 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 		// Ensure all input blocks are valid.
 		stats, err := block.GatherIndexHealthStats(cg.logger, filepath.Join(bdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
 		if err != nil {
-			return false, ulid.ULID{}, errors.Wrapf(err, "gather index issues for block %s", bdir)
+			level.Error(cg.logger).Log("msg", "gather index issues for block ", bdir, err)
```
Comment (bwplotka) [inline pkg/compact/compact.go:731]: In the local scope, this fix is great, but unfortunately, if we don't deal with this and just ignore it, the next compaction will pick the same set of blocks (: 

Let's think about automated solution to this problem. Ideally while changing compactor algorithm: https://github.com/thanos-io/thanos/issues/4046#issuecomment-823373171

**Review by bwplotka — CHANGES_REQUESTED** [review]
Thanks for this attempt, but we need to invest in different planning and mechanisms that allows this as commented as https://github.com/thanos-io/thanos/issues/4046#issuecomment-823373171 

Hope that makes sense 🤗

**Comment by mikechengwei** [comment]
/

**Pull request closed without merging**